httpcore5/src/main/java/org/apache/hc/core5/net/URIBuilder.java

```diff
@@ -499,6 +499,16 @@ public URIBuilder setPath(final String path) {
         return this;
     }
 
+    /**
+     * Appends path to URI. The value is expected to be unescaped and may contain non ASCII characters.
+     *
+     * @return this.
+     */
+    public URIBuilder appendPath(final String path) {
```

Comment:
@larshelge One last thing. I suppose this should be a little more efficient by avoiding unnecessary operations and intermediate garbage.
```
    public URIBuilder appendPath(final String path) {
        if (path != null) {
             appendPathSegments(splitPath(path);
        }
        return this;
    }
```

Reply:
Updated it now.